Added note about TLS bare IP address limitation (#174)
bytebeamio/rumqtt: rumqttc/README.md

In short, everything necessary to maintain a robust connection

Since the eventloop is externally polled (with `iter()/poll()` in a loop)
out side the library and `Eventloop` is accessible, users can
- Distribute incoming messages based on topics
- Stop it when required
- Access internal state for use cases like graceful shutdown or to modify options before reconnection

## Important notes

> - Looping on `connection.iter()/eventloop.poll()` is necessary to run the

Reviewer: Separate these because they are two different pieces of code.

```suggestion
- Looping on `connection.iter()`/`eventloop.poll()` is necessary to run the
```

  event loop and make progress. It yields incoming and outgoing activity
  notifications which allows customization as you see fit.

> - Blocking inside the `eventloop.poll()/eonnection.iter()` loop will block

Reviewer: There is a typo in "connection" here, but I suggest you switch the two just to be consistent with the note above.

```suggestion
- Blocking inside the `connection.iter()`/`eventloop.poll()` loop will block
```

>   connection progress

Reviewer: All other points end with a dot.

```suggestion
  connection progress.
```

- You cannot create a TLS connection to a bare IP address with a self-signed

>   certificate. This is a [limitation of Rustls](https://github.com/ctz/rustls/issues/184).

Reviewer: The name "rustls" isn't capitalized in the official documentation (except for the beginning of a sentence, of course).

```suggestion
  certificate. This is a [limitation of rustls](https://github.com/ctz/rustls/issues/184).
```

Author: Thanks, I actually did only spot it at the beginning of sentences.

  One workaround, which only works under *nix/BSD-like systems, is to add an

>   entry to wherever your DNS resolver looks (eg. `/etc/hosts`) for the bare IP

Reviewer: I feel like "e.g." is more correct than "eg." - being an abbreviation for "exempli gratia" -, but I looked it up and there doesn't seem to be a consensus on this, so it's just my opinion, really - feel free to ignore.

```suggestion
  entry to wherever your DNS resolver looks (e.g. `/etc/hosts`) for the bare IP
```

Author: The rule is basically a matter of style nowadays, rather than correctness. Happy to change it.
